### ISSUE TITLE
bending energy deformed loss

### ISSUE THREAD

**Issue by kate-sann5100** [Project-MONAI/MONAI#1405]
**Is your feature request related to a problem? Please describe.**
Bending energy is a common loss used for image registration.

**Describe the solution you'd like**
Implement bending energy in `losses/deform.py`

**Describe alternatives you've considered**
N/A

**Additional context**
[Tensorflow implementation of bending energy by DeepReg](https://github.com/DeepRegNet/DeepReg/blob/f29b1c658124c8d3a5f216662a0a27d73212917e/deepreg/model/loss/deform.py#L77-L102)

**Comment by tvercaut** [comment]
Note: #1417 span out from the implementation of https://github.com/Project-MONAI/MONAI/blob/master/monai/losses/deform.py

**Comment by YipengHu** [comment]
@kate-sann5100 one thing worth investigating here - because the pytorch implementation of the resampler, grid_sample, requires normalising grid locations to [-1,1], this may have an effect on computing the gradient (and 2nd derivatives) when using finite difference. This will make the loss on deformation dependent on the size of the input (actually the ratio between H, W and D). I'd say to normalise it by dividing the image shape by 1/H, 1/W and 1/D, instead of 2. to be consistent with the unit of voxel. A grander thing to do is allowing anisotropic voxel dimensions (resolution) and accepting the physical image voxel sizes in mm/voxel - but this may require a much bigger change.  
thoughts?

**Comment by kate-sann5100** [comment]
Here is a conclusion of me and @YipengHu 's off-line discussion.

Firstly, the normalisation is applied only in the [Warp layer](https://github.com/Project-MONAI/MONAI/blob/bcdee8c062031a75e95cd777d7108d9db31970e2/monai/networks/blocks/warp.py#L105) and do not affect the computation of bending energy. 

Secondly, the normalisation is dependent on input volume shape, but the effect should be compensated in pytorch `grid_sample`.

Based on the above reasons, we believe there should not be a problem in the current implementation of deformation loss. For the Warp layer, a comparison between performance of the pytorch `gird_sample` and MONAI's 'grid_pull' would be helpful as a double-check.

@wyli

**Comment by wyli** [comment]
thanks for the update. so this ticket is addressed by https://github.com/Project-MONAI/MONAI/pull/1408 I'm closing it
